### PR TITLE
State transition

### PR DESCRIPTION
**Pull request by joaobrittoneto**
Do runtime state transition in a virtual method. 
Each task defines its runtime state independently. 

Make MISSING_TRANSFORMATION a exception

### CODE REVIEW

File: uwv_dynamic_model.orogen [hunk 1]
```diff
@@ -121,6 +121,6 @@ task_context "VelocityEstimator" do
         max_latency 0.05
     end
 
-    runtime_states :MISSING_TRANSFORMATION
+    exception_states :MISSING_TRANSFORMATION
```
Comment (doudou) [inline uwv_dynamic_model.orogen]: You cannot make MISSING_TRANSFORMATION an exception state. With the way the transformer is currently implemented, it is expected in the few first samples, as the transformer "discovers" the transformation streams.

Comment (joaobrittoneto) [inline uwv_dynamic_model.orogen]: Ok, I was thinking it was like a property, at least for a static transformation.

Comment (joaobrittoneto) [inline uwv_dynamic_model.orogen]: Commit removed

File: tasks/Task.hpp [hunk 2]
```diff
@@ -75,6 +75,12 @@ namespace uwv_dynamic_model {
 	virtual void handleStates(const base::samples::RigidBodyState &state, const base::LinearAngular6DCommand &control_input);
 
 	/**
+	 * Set RunTime component state idepently
```
Comment (rafaelsaback) [inline tasks/Task.hpp]: "independently"

File: uwv_dynamic_model.orogen (hunk 1 above)
Comment (doudou) [inline uwv_dynamic_model.orogen]: When using static transforms, it's correct. For dynamic transforms, it's not.

When I get the time to implement my plans for transformer++, this situation will never happen again anyways ...

File: tasks/Task.hpp (hunk 2 above)
Comment (joaobrittoneto) [inline tasks/Task.hpp]: Ops.
Corrected

**Comment by doudou** [comment]
Looks good to me, @rafaelsaback ?

**Comment by rafaelsaback** [comment]
Good to go @joaobrittoneto :+1:

**Pull request merged**